packages/legend-studio/src/stores/editor-state/ModelLoaderState.ts

```diff
@@ -164,6 +195,10 @@ export class ModelLoaderState extends EditorState {
             this.currentExternalInputType,
             ImportMode.SCHEMA_IMPORT,
           )) as Entity[];
+      } else if (this.currentExtensionInputType) {
+        entities = (yield this.currentExtensionInputType.load(
```

Comment:
I think we should return after this load. The extension should decide how to load the entities. Below we do a hard save through `updateEntities` which we may not want to do for all the plugins.